llvm/lib/Transforms/Yk/StackMaps.cpp

```diff
@@ -61,7 +61,15 @@ class YkStackmaps : public ModulePass {
             // emit a stackmap in those cases too.
             if (!CI.isIndirectCall() && CI.getCalledFunction()->isIntrinsic())
               continue;
-            SMCalls.insert({&I, LA.getLiveVarsBefore(&I)});
+            std::vector<Value *> Lives = LA.getLiveVarsBefore(&I);
+            // Value guards for variable promotions, require us to also
+            // track the value generated by the call.
+            if (CI.getCalledFunction() &&
+                CI.getCalledFunction()->getName() == "yk_promote") {
```

Comment:
`yk_promote` is user-facing, so I prefer not to give them double underscored names.

Reply:
The *user* should still see `yk_promote`. I'm saying we should do the following in C:

```
#define yk_promote(x) __yk_promote(x)
```

because if (and I know it's unlikely! but it is more standard) they define their own function called `yk_promote` we won't accidentally match it here. Basically I'm suggesting user-invisible name mangling.

Reply:
If they have a function called "yk_promote" and we do your `#define` thing, then we divert all of the calls to their `yk_promote` from theirs to ours.

There's no getting around it :P

Reply:
> If they have a function called "yk_promote" and we do your #define thing, then we divert all of the calls to their yk_promote from theirs to ours.

Only when they include our header!

Reply:
OK, so why do you think it is not common place to do this?

Why are all of the function (e.g. in gtk, like `gtk_application_new()`) not `#defined` like this? Just in case the user is strapping a GUI onto a program that already has a `gtk_application_new()`?

Reply:
We agreed to disagree.

e6f65fc2471ca994a13df65a126711ab3a8ff935 and a runtime commit also: https://github.com/ykjit/yk/pull/750/commits/00318e623cfb8fc9c6f54d35669163ac3142a486

Reply:
α-renaming is good for the soul -- and the debugger :) Thanks!